rocks-lib/src/config/mod.rs

```diff
@@ -173,8 +185,16 @@ impl Default for Config {
             only_sources: None,
             namespace: "".into(),
             lua_dir: data_path.join("lua"),
-            lua_version: None,
-            tree: data_path,
+            lua_version: current_project
+                .as_ref()
+                .map(|project| LuaVersion::try_from(project.rockspec()))
+                .transpose()
+                .unwrap(),
+            tree: current_project
+                .as_ref()
+                .map(|project| project.root().join("tree"))
```

Comment:
issue: this needs to be moved out into a different place. You can't test for `--no-project` in time here, as it'll always be `false` in the default configuration.